include/mqtt/sync_client.hpp

```diff
@@ -46,7 +52,13 @@ class sync_client : public client<Socket, PacketIdBytes> {
      * @param port port number
      * @return sync_client object
      */
-    friend std::shared_ptr<callable_overlay<sync_client<tcp_endpoint<as::ip::tcp::socket, null_strand>>>>
+    friend std::shared_ptr<callable_overlay<sync_client<tcp_endpoint<as::ip::tcp::socket,
+#if defined(MQTT_NO_TS_EXECUTORS)
+    as::io_context::executor_type
+#else
+    null_strand
+#endif
+    >>>>
```

Comment:
null_strand is a kind of null object pattern.
If has the same interface as strand but do nothing. 
So if user wants to use mqtt_cpp on single thread without locks, the user can use it.
What happens as::io_context::executor_type ?
Does this work as null_strand?

Reply:
Could you use the following coding style for multi line template ?
https://github.com/redboltz/mqtt_cpp/blob/117155f8ee803f5bd046d05970c98d59a2617a52/include/mqtt/v5_message.hpp#L546-L552
See also https://github.com/redboltz/mqtt_cpp/wiki/Coding-Rules

In this case, please use the following indent.

```cpp
    friend std::shared_ptr<
        callable_overlay<
            sync_client<
                tcp_endpoint<
                    as::ip::tcp::socket,
#if defined(MQTT_NO_TS_EXECUTORS)
                    as::io_context::executor_type
#else
                    null_strand
#endif
                >
            >
        >
    >
```

Reply:
Sorry, I missed the part about template indention in the Coding Guidelines.
I'll update my code accordingly.

The proposed standard executor style of Asio provides executors per context that can be copied and modified to track outstanding work - not exit `ctx.run()` when nothing has been queued - for example.
The `as::io_context::executor_type` contains a pointer to the context it is associated with.
Like this it works the same as _null_strand_ did before but uses a copy of the `ctx.get_executor()` directly instead of going through an additional object.

Reply:
> Sorry, I missed the part about template indention in the Coding Guidelines.

I updated the wiki **after you post the PR**.  So you didn't miss anything at that time :)

> I'll update my code accordingly.

Thank you.

> The proposed standard executor style of Asio provides executors per context that can be copied and modified to track outstanding work - not exit `ctx.run()` when nothing has been queued - for example.
> The `as::io_context::executor_type` contains a pointer to the context it is associated with.
> Like this it works the same as _null_strand_ did before but uses a copy of the `ctx.get_executor()` directly instead of going through an additional object.

Good to know. I understand that the proposed standard version already has null_strand like mechanism.